libsrc/atari/getdefdev.s

```diff
@@ -68,16 +69,23 @@ crvec:  jsr     $FFFF           ; will be set to crunch vector
         sta     __defdev
         iny
         lda     (DOSVEC),y
-        sta     __defdev+1
+done:   sta     __defdev+1
 
 ; Return pointer to default device
 
 finish: lda     #<__defdev
         ldx     #>__defdev
         rts
 
+; XDOS version
+
+xdos:   lda     XDEFDEV
+        bne     done
+
         .data
 
+crvec:	.byte	$4C,$FF,$FF     ; will be set to crunch vector
```

Comment:
Trying to add a more helpful comment:

The idea is to use mnemonics everywhere the generated bytes are fetched/executed by the CPU and .byte everywhere the generated bytes are read/written by the CPU.